### ISSUE TITLE
google_compute_autoscaler:  Error 400: Required field 'autoscaler' not specified

### ISSUE THREAD

**Issue by mattes** [hashicorp/terraform#15056]
### Terraform Version
Terraform v0.9.6

### Affected Resource(s)
- google_compute_autoscaler

### Terraform Configuration Files
```hcl
resource "google_compute_autoscaler" "primary" {
  name   = "primary"
  zone   = "${google_container_cluster.primary.zone}"
  target = "${google_compute_instance_group_manager.primary.self_link}"

  autoscaling_policy = {
    max_replicas    = 5
    min_replicas    = 4
    cooldown_period = 60

    cpu_utilization {
      target = 0.75
    }
  }
}
```

### Output
```
google_compute_autoscaler.primary: Error updating Autoscaler: googleapi: Error 400: Required field 'autoscaler' not specified, required
```

**Comment by danielcompton** [comment]
That autoscaler looks almost identical to my working config. Can you share more of your config, I suspect there is something else that is going on here?

**Comment by mattes** [comment]
```hcl
resource "google_container_cluster" "primary" {
  name               = "primary"
  zone               = "us-central1-a"
  initial_node_count = 3

  master_auth {
    username = "${var.cluster_master_auth_username}"
    password = "${var.cluster_master_auth_password}"
  }

  node_config {
    machine_type = "n1-standard-1"
    disk_size_gb = 100
  }
}


resource "google_compute_instance_group_manager" "primary" {
  name               = "${var.google_compute_instance_group_manager_primary_name}"
  zone               = "${google_container_cluster.primary.zone}"
  instance_template  = "https://www.googleapis.com/compute/v1/projects/${var.google_project_id}/global/instanceTemplates/${replace(var.google_compute_instance_group_manager_primary_name, "/-grp$/", "")}"
  base_instance_name = "${replace(var.google_compute_instance_group_manager_primary_name, "/-grp$/", "")}"

  target_size = 3
  named_port {
    name = "http-to-https-port"
    port = 31000
  }
}
```

If it's my config, maybe the `target_size` is causing problems?

**Comment by mattes** [comment]
Just tested without `target_size`. Same error.

**Comment by rileykarson** [comment]
Hi @mattes,
The error isn't in your config - it's because Terraform is trying to perform an update, but that functionality is broken right now. You can work around this for now by changing the name to force a new resource, or by manually running `terraform destroy` with your resource as a `-target` parameter, followed by `terraform apply`.

Note that both approaches will involve destroying and recreating the resource - if you have other resources that depend on it, you may want to manually edit it in the Cloud Console/with the gCloud API until updating works properly again.

**Comment by danielcompton** [comment]
For completeness, you can also use 

```hcl
lifecycle {
  create_before_destroy = true
}
```

although I'm not sure how well that will work with downstream resources.

**Comment by matschaffer** [comment]
For anyone who lands here like I did, looks like a proper fix for this went out in [0.9.7](https://github.com/hashicorp/terraform/blob/master/CHANGELOG.md#097-june-7-2017)
